src/app/redux/TransactionReducer.js

```diff
@@ -0,0 +1,187 @@
+export const BROADCAST_OPERATION = 'transaction/BROADCAST_OPERATION';
+export const UPDATE_AUTHORITIES = 'transaction/UPDATE_AUTHORITIES';
+export const UPDATE_META = 'transaction/UPDATE_META';
+const ERROR = 'transaction/ERROR';
+const DELETE_ERROR = 'transaction/DELETE_ERROR';
+const SET = 'transaction/SET';
+const REMOVE = 'transaction/REMOVE';
+// Saga-related
+export const RECOVER_ACCOUNT = 'transaction/RECOVER_ACCOUNT';
+
+const defaultState = fromJS({
+    operations: [],
+    status: { key: '', error: false, busy: false, },
+    errors: null
+});
+
+export default function reducer(state = defaultState, action) {
+    const payload = action.payload;
+
+    switch (action.type) {
+        case 'transaction/CONFIRM_OPERATION': {
+            const operation = fromJS(payload.operation);
+            const confirm = payload.confirm;
+            const warning = payload.warning;
+            return state.merge({
+                show_confirm_modal: true,
+                confirmBroadcastOperation: operation,
+                confirmErrorCallback: payload.errorCallback,
+                confirm,
+                warning
+            });
+        }
+
+        case 'transaction/HIDE_CONFIRM':
+            return state.merge({show_confirm_modal: false, confirmBroadcastOperation: undefined, confirm: undefined});
+
+        case 'transaction/BROADCAST_OPERATION':
+            // See TransactionSaga.js
+            return state;
+
+        case 'transaction/UPDATE_AUTHORITIES':
+            return state;
+
+        case 'transaction/UPDATE_META':
+            return state;
+
+        case 'transaction/ERROR': {
+            const {operations, error, errorCallback} = payload;
+            let errorStr = error.toString();
+            let errorKey = 'Transaction broadcast error.';
+            for (const [type/*, operation*/] of operations) {
+                switch (type) {
+                    case 'vote':
+                        if (/uniqueness constraint/.test(errorStr)) {
+                            errorKey = 'You already voted for this post';
+                            console.error('You already voted for this post.')
+                        }
+                        break;
+                    case 'comment':
+                        if (/You may only post once per minute/.test(errorStr)) {
+                            errorKey = 'You may only post once per minute.'
+                        } else if (errorStr === 'Testing, fake error')
+                            errorKey = 'Testing, fake error';
+                        break;
+                    case 'transfer':
+                        if (/get_balance/.test(errorStr)) {
+                            errorKey = 'Insufficient balance.'
+                        }
+                        break;
+                    case 'withdraw_vesting':
+                        if(/Account registered by another account requires 10x account creation fee worth of Steem Power/.test(errorStr))
```

Comment:
This regex isn't going to change. It should be a const with a nice descriptive name.